### DISCUSSION TITLE
Icons Update

### DISCUSSION THREAD

**Discussion by rexelbartolome** [freeplane/freeplane#285] [Show and tell]
Here's what's been happening with the icons lately :D

https://user-images.githubusercontent.com/88552647/160814302-c52d6aa9-a623-4968-b3c9-f2de04999e2c.mp4

https://user-images.githubusercontent.com/88552647/160813896-8597f99b-5d07-4558-a900-64ba68bd3f27.mp4

All the icons for the dropdown and various menus are done (thanks to Predrag!), the others that haven't been replaced are soon to be removed. If anything, FP has more than enough icons 😆

Currently, we just need to handle how to implement it with freeplane.properties but so far it looks great on different LaF's with its responsive color changes 

![javaw_u7S3hBDjtX](https://user-images.githubusercontent.com/88552647/160814634-71118b7b-89d0-43ee-b237-fa9936a74adc.png)
![javaw_7Y1UcJmzkg](https://user-images.githubusercontent.com/88552647/160814638-cbdc2dc8-6334-44e6-83a2-4aeeda4d90ce.png)
![javaw_doRMLxWNf2](https://user-images.githubusercontent.com/88552647/160814643-b19921a0-b918-4706-a72d-f306e9165eaf.png)

**Comment by macmarrum** [comment]
Thanks for the update, Rexel.
The overall look with the new icons is great :+1:

**Comment by EdoFro** [comment]
Great work, Rexel!!

Please, don't forget to change the **attribute's** and the **note's** icons too.

I use custom icons for both and mindmaps look a lot cleaner:

![image](https://user-images.githubusercontent.com/35700575/160866167-6154e122-afd5-442b-969e-8bc28e6fbcd3.png)

**Reply by EdoFro** [reply]
the two svg files:

[showAttributes.svg](https://user-images.githubusercontent.com/35700575/160867703-d1a4f424-0ce7-463a-b191-3921506ea88d.svg)
[knotes.svg](https://user-images.githubusercontent.com/35700575/160867706-bdb78394-735f-4e88-9d14-85135b40e4e4.svg)

**Reply by PredragC** [reply]
I made icons similar to yours but in line with the other icons we use for attributes and notes, please confirm if this is ok for you.
![N A](https://user-images.githubusercontent.com/102694585/160925520-1a6b509a-caba-4ae9-8767-7ea614eb9788.png)

**Reply by EdoFro** [reply]
perfect, as long as they are slim and simple, it's ok to me. thanks!

**Reply by dpolivaev** [reply]
I have recently tried the new icons. Personally, I do not like the new "with Note" icon. For me, it looks like a bug, like an unexpected blue line to the left of the text. I could associate the old note icon which you also can see in the tooltip with the note easily, but not the new one. @PredragC could you create something different?

**Comment by dpolivaev** [comment]
The latest preview version 1.10.1_04 contains all new icons. 
Kudos to @rexelbartolome and @PredragC.

https://sourceforge.net/projects/freeplane/files/freeplane%20preview/

Technically I could include the new icons also into 1.9.x version but I don't see why.

**Comment by nbruno78** [comment]
Hi,
I know that a lot of work has been performed on these icons, and I would like to thank all involved people for this. But personnally I enjoyed the variety of colours of the old icon scheme, in line with @sifran remark from 2022-01-15  on the old forum (https://sourceforge.net/p/freeplane/discussion/758437/thread/8b6f0779e8/ ) : 
_"I really don't like the change.... I don't like fact that all the icons have the same color scheme. it makes everything looks the same, and its hard to find the actions."_
Unfortunately colours depend on cultures ; for me blue=sad, blue+black = sad+sad... and all square shaped icons (even the padlock is square) add rigidity to the sadness.
I hope that my remark does not hurt anybody (it is not the purpose), and maybe I am the only one who feels this. Or maybe not.

**Reply by EdoFro** [reply]
Hi,
I think it's important for Freeplane to renew its user interface. It should look cleaner and up to date. Until a few months it looked old and outdated.
This is important if we want that this software has a longer life and a growing user base.

That been said, I understand your position and point of view. Luckily it's relative easy to have the icons you like showing in Freeplane. When you put an svg file in your user directory that the same name than the original icon, then Freeplane uses that icon and not the one that comes with the installed version of Freeplane.

The folder is: `<userDirectory>/resources/images/`

If you can copy the old svg files to that folder you will retain the old icons and they won't change when you update to newer Freeplane versions.

**Reply by EdoFro** [reply]
Here is a zip file containing the "old" icons. Just unzip it and put the files into the `<userDirectory>/resources/images/` folder.

Hope it helps.

[icons FP 1.9.14.zip](https://github.com/freeplane/freeplane/files/8418922/icons.FP.1.9.14.zip)

**Reply by EdoFro** [reply]
Making an add-on, that installs the old icons in the user directory, is an easy task, if somebody wants to give it a try.

I'm busy now to do it myself, but I can help answering any doubts.

**Reply by abc16361** [reply]
> Here is a zip file containing the "old" icons. Just unzip it and put the files into the `<userDirectory>/resources/images/` folder.
> 
> Hope it helps.
> 
> [icons FP 1.9.14.zip](https://github.com/freeplane/freeplane/files/8418922/icons.FP.1.9.14.zip)

It does not replace all of them. Do some new icons use different names? Some new icons do look similar and are not easy to tell the difference, particularly in small size.  I would like to suggest keeping the old icons as default and let users to choose their favourite icons in user directory.

![icons](https://user-images.githubusercontent.com/101362122/161893847-e86fead4-d315-4495-869e-98effc08ad5f.png)

**Reply by rexelbartolome** [reply]
Hmmmm I'm not sure how to help honestly :( I do wish that people that prefer the old icons could have some sort of choice, but with how the change was implemented, the freeplane.properties was also changed to have a better naming scheme to refer to some of the icons. So some of the old icons would need to be renamed to work properly. And the color scheme has been made like this (one color for everything) so that it's more versatile when a differently colored LaF is applied.

If you're willing, it's a lengthy process of going to opening freeplaneviewer.jar, opening freeplane.properties, looking at which **actions** refer to which **icons**, then searching and renaming the appropriate **icon** for that **action**.

Other than that, maybe you can use an older freeplane.properties (or maybe just parts of it) and put it inside your freeplaneviewer.jar so it has the old naming scheme (so no renaming required) but it might break Freeplane in another way

BTW, you can change the icon size to be bigger, that might help with readability

![image](https://user-images.githubusercontent.com/88552647/161903239-d9a558ef-a18a-4061-8347-40f66c053920.png)

But the blue accent color can definitely be changed as it depends on what Look and Feel is active. The current selection of Flat LaF's are all found inside this collection https://www.formdev.com/flatlaf/ Opening this demo will show you a menu of the different color schemes.

![image](https://user-images.githubusercontent.com/88552647/161900342-0f221c05-d274-4293-a3c3-f420137534cf.png)

https://user-images.githubusercontent.com/88552647/161900455-53c268cf-8c41-4e51-8496-0677e8fc2092.mp4

Then after that,
you can suggest to @dpolivaev what LaF could be a great addition. Maybe Arc Orange could suit you better?


![image](https://user-images.githubusercontent.com/88552647/161901003-a2bbbbcd-d593-4279-b0e8-9a61f6183c3c.png)

So in conclusion:

Adding multiple colors for different icons is unlikely to be implemented unless other contributors could somehow make it work and be readable in both light and dark themed LaFs.

The accent color of the icons can be changed to a different color depending on the LaF theme color

The old icons can still be used if you're willing to hack the freeplane.properties and resources folder to a significant degree

**Reply by abc16361** [reply]
@rexelbartolome 
Thanks!
I think I would use the new icons set. For those new icons that look similar I think I could get used to them after some practice, or decorate them with colours if I get picky. What I suggested using user directory is just a feature that could be easier for users to choose an icon set, where different icon sets are hold and icons can be replaced. 

The new themes of Flat LaF's are good. I like Flat Light and Flat IntelliJ. I tried them but give it up reluctantly because of some scripts I am using. I would create a thread about it later if I know what I should talk.

**Reply by dpolivaev** [reply]
It wouldn't be hard to allow replaceable sets of icons. However, maintaining multiple sets of actions is work in the long term. because new actions might come in, some files could be renamed and some properties changed. Unless someone commits to being a maintainer I would not recommend trying to use the old set in new versions.

> "I really don't like the change.... I don't like fact that all the icons have the same color scheme. it makes everything looks the same, and its hard to find the actions."

I somehow feel the pain. I agree that the old icons were more diverse in their colors and shapes. And for me, it looked like a conscious decision to go the minimalistic way made by our artists. As a consequence, many icons have no own background color and therefore their accent colors must depend on the look and feel. It wouldn't be necessary otherwise. I also do not agree that the wish to make the colors follow the look and feel own colors enforces any limitations in the number of colors used in the icons because the current code allows to use of any number of look and feel depended colors.

But therefore all changes underwent a community discussion. We got the first draft developed by @rexelbartolome, it got mostly positive feedback, and then the decision was made. And then the artist has their creative freedom.

**Reply by EdoFro** [reply]
> The new themes of Flat LaF's are good. I like Flat Light and Flat IntelliJ. I tried them but give it up reluctantly because of some scripts I am using. I would create a thread about it later if I know what I should talk.

I'm having some issues too, with scripts that display dialogs.

**Comment by Illegit** [comment]
I feel the need to thank Bartolome (i like your icons: more professional) for the new icons and EdoFro to providing a clear and simple explanation to meet the needs expressed by nbruno78 (in my case: in some way it could be useful to have the opportunity to change them sometimes (audience? impact))
Bartolome, there are still some icons not converted do you think to approach them? (filter&, filter or, node attributes)
Regards
SFPC

**Reply by rexelbartolome** [reply]
Yep we're aware, the icons are already done for them and Dimitry has implemented it. But the preview release isn't out for it yet, I think he'll be able to release another one this weekend.

**Reply by dpolivaev** [reply]
Yes, it's the plan

**Comment by Illegit** [comment]
Thank you.
SFPC

**Comment by Zorco1** [comment]
Personnaly, although a great work has been done, I still regret colored icons.

I think that's the exact opposite of the mindmapping philosophy that requires use of colors everywherere we can to add information and use all our brain capacity and the mindmapping founder (R. Buzan) would have an heart attack if he could see the flat menu. It's a pitty, our world is rejecting colors everywhere only by mode.

https://mindmapsunleashed.com/how-to-mind-map-with-tony-buzan
use colors

In future realeases I suggest to reconsider the use  of colors by default and flat as a option for all people interested.
